internal/logger/log.go

```diff
@@ -5,223 +5,160 @@ SPDX-License-Identifier: AGPL-3.0-only
-func (l *Logger) GetServerStreamInterceptor() grpc.ServerOption {
+func GetServerStreamInterceptor(l *slog.Logger) grpc.ServerOption {
 	return grpc.StreamInterceptor(
-		logging.StreamServerInterceptor(l.middlewareLogger()),
+		logging.StreamServerInterceptor(middlewareLogger(l)),
 	)
 }
 
 // GetClientUnaryInterceptor returns a gRPC client option for intercepting unary gRPC logs.
-func (l *Logger) GetClientUnaryInterceptor() grpc.DialOption {
+func GetClientUnaryInterceptor(l *slog.Logger) grpc.DialOption {
 	return grpc.WithUnaryInterceptor(
-		logging.UnaryClientInterceptor(l.middlewareLogger()),
+		logging.UnaryClientInterceptor(middlewareLogger(l)),
 	)
 }
 
 // GetClientStreamInterceptor returns a gRPC client option for intercepting stream gRPC logs.
-func (l *Logger) GetClientStreamInterceptor() grpc.DialOption {
+func GetClientStreamInterceptor(l *slog.Logger) grpc.DialOption {
 	return grpc.WithStreamInterceptor(
-		logging.StreamClientInterceptor(l.middlewareLogger()),
+		logging.StreamClientInterceptor(middlewareLogger(l)),
 	)
 }
 
-// getZapLogger returns the underlying zap logger.
-func (l *Logger) getZapLogger() *zap.Logger {
-	return l.logger.Desugar()
-}
-
-func (l *Logger) middlewareLogger() logging.Logger {
+func middlewareLogger(l *slog.Logger) logging.Logger {
 	return logging.LoggerFunc(func(ctx context.Context, lvl logging.Level, msg string, fields ...any) {
-		f := make([]zap.Field, 0, len(fields)/2)
+		f := make([]slog.Attr, 0, len(fields)/2)
 
 		for i := 0; i < len(fields); i += 2 {
 			key := fields[i]
 			value := fields[i+1]
 
 			switch v := value.(type) {
 			case string:
-				f = append(f, zap.String(key.(string), v))
+				f = append(f, slog.String(key.(string), v))
 			case int:
-				f = append(f, zap.Int(key.(string), v))
+				f = append(f, slog.Int(key.(string), v))
 			case bool:
-				f = append(f, zap.Bool(key.(string), v))
+				f = append(f, slog.Bool(key.(string), v))
 			default:
-				f = append(f, zap.Any(key.(string), v))
+				f = append(f, slog.Any(key.(string), v))
 			}
 		}
 
-		logger := l.getZapLogger().WithOptions(zap.AddCallerSkip(1)).With(f...)
+		var pcs [1]uintptr
+		runtime.Callers(2, pcs[:]) // skip [Callers, LoggerFunc]
+		r := slog.Record{}
 
 		switch lvl {
 		case logging.LevelDebug:
-			logger.Debug(msg)
+			r = slog.NewRecord(time.Now(), slog.LevelDebug, fmt.Sprintf(msg, fields...), pcs[0])
 		case logging.LevelInfo:
-			logger.Info(msg)
+			r = slog.NewRecord(time.Now(), slog.LevelInfo, fmt.Sprintf(msg, fields...), pcs[0])
 		case logging.LevelWarn:
-			logger.Warn(msg)
+			r = slog.NewRecord(time.Now(), slog.LevelWarn, fmt.Sprintf(msg, fields...), pcs[0])
 		case logging.LevelError:
-			logger.Error(msg)
+			r = slog.NewRecord(time.Now(), slog.LevelError, fmt.Sprintf(msg, fields...), pcs[0])
 		default:
 			panic(fmt.Sprintf("unknown level %v", lvl))
 		}
+		_ = l.Handler().Handle(context.Background(), r)
```

Comment:
We can already panic above, so I think we should panic here too if it throws an error. If there's other reasoning behind this, we need to add a [nolint directive](https://golangci-lint.run/usage/false-positives/#nolint-directive) here.

Reply:
It's common to ignore errors relating to log or console output.
The panic above is a logic error, while the panic here would be a runtime error.
Is `_ = ` not sufficient to be ignored by the linter?

Reply:
What is the consensus here? I am in favor of panicking as few times as possible.

Reply:
imho, we shouldn't panic in both cases. As TT said, the underscore assignment is probably sufficient to satisfy the linter. @thomasten what do you think regarding the panics?

Reply:
Seeing that it may be that services pass levels into this function that aren't handled by `slog` out of the box and would have to be added ourselves (for example with a [replaceAttr function](https://pkg.go.dev/log/slog#example-HandlerOptions-CustomLevels)) I would argue that we should go with the last proposal and log on a hardcoded level in default. If we just cast the level I believe `slog` logs that as `<level>+<diff-to-int>`, for example casting 2 would result in the `slog` level "INFO+2", and I don't know how searchable that is.

Reply:
Agreed